src/realm/object-store/list.hpp

```diff
@@ -219,35 +246,80 @@ void List::set(Context& ctx, size_t list_ndx, T&& value, CreatePolicy policy)
         ctx.template unbox<Obj>(value, policy, key);
         return;
     }
-    dispatch([&](auto t) {
-        this->set(list_ndx, ctx.template unbox<std::decay_t<decltype(*t)>>(value, policy));
-    });
+    if (m_type == PropertyType::Mixed) {
+        Mixed new_val = ctx.template unbox<Mixed>(value, policy);
+        if (new_val.is_type(type_Dictionary)) {
+            set_collection(list_ndx, CollectionType::Dictionary);
```

Comment:
There appears to be a bunch of duplicated code between List and Dictionary.

Reply:
Similar, but not the same.